rbac/migration_tool/migrate.py

```diff
@@ -26,109 +26,89 @@
-            # These should come from the apps that own the resource.
-            if (
-                bound_resource.resource_type == ("rbac", "workspace")
-                and not bound_resource.resourceId == default_workspace
-            ):
-                # This is not strictly necessary here and the relation may be a duplicate.
-                # Once we have more Workspace API / Inventory Group migration progress,
-                # this block can and probably should be removed.
-                # One of those APIs will add it themselves.
-                relationships.append(
-                    create_relationship(
-                        bound_resource.resource_type,
-                        bound_resource.resourceId,
-                        ("rbac", "workspace"),
-                        default_workspace,
-                        "parent",
-                    )
-                )
-
+        bound_resource = v2_role_binding.resource
+
+        # Is this a workspace binding, but not to the root workspace?
+        # If so, ensure this workspace is a child of the root workspace.
+        # All other resource-resource or resource-workspace relations
+        # which may be implied or necessary are intentionally ignored.
+        # These should come from the apps that own the resource.
+        if (
+            bound_resource.resource_type == ("rbac", "workspace")
+            and not bound_resource.resource_id == default_workspace
+        ):
+            # This is not strictly necessary here and the relation may be a duplicate.
+            # Once we have more Workspace API / Inventory Group migration progress,
+            # this block can and probably should be removed.
+            # One of those APIs will add it themselves.
             relationships.append(
                 create_relationship(
                     bound_resource.resource_type,
-                    bound_resource.resourceId,
-                    ("rbac", "role_binding"),
-                    v2_role_binding.id,
-                    "user_grant",
+                    bound_resource.resource_id,
+                    ("rbac", "workspace"),
+                    default_workspace,
+                    "parent",
                 )
             )
 
-    return relationships, binding_mappings
+        relationships.append(
+            create_relationship(
+                bound_resource.resource_type,
+                bound_resource.resource_id,
+                ("rbac", "role_binding"),
+                v2_role_binding.id,
+                "user_grant",
+            )
+        )
+
+    return relationships
 
 
 def migrate_role(
     role: Role,
     write_relationships: bool,
     default_workspace: str,
-    current_mapping: Optional[BindingMapping] = None,
-) -> tuple[list[common_pb2.Relationship], BindingMappings]:
+    current_bindings: Iterable[BindingMapping] = [],
+) -> tuple[list[common_pb2.Relationship], list[BindingMapping]]:
```

Comment:
This now returns a list of `BindingMapping`s because a Role creates multiple role bindings, and each role binding gets its own Mapping object so that they can be locked individually. This is specifically to better support storing the mappings for system roles, because those roles will have 10-100k associated mappings which we didn't want to store on a single object.